src/network/networkd-route.c

```diff
@@ -1868,6 +1868,31 @@ int config_parse_route_scope(
         return 0;
 }
 
+int route_table_from_string_full(Network *network, const char *s, uint32_t *ret) {
+        int r;
+
+        assert(s);
+        assert(network);
+        assert(ret);
+
+        r = route_table_from_string(s);
+        if (r >= 0)
+                *ret = (uint32_t) r;
+        else {
+                if (!hashmap_isempty(network->manager->route_tables)) {
```

Comment:
If the hashmap is not empty, then the setting cannot accept integer... So, please drop `hashmap_isempty()` and do like the following:
```
uint32_t t = PTR_TO_UINT32(hashmap_get(...));
if (t != 0) {
    *ret = t;
    return 0;
}

return safe_atou32(s, ret);
```